### DISCUSSION TITLE
Documentation feedback?

### DISCUSSION THREAD

**Discussion by hx2A** [py5coding/py5generator#285] [General]
Please have a look at this draft documentation on the py5 dev website:

http://py5dev.ixora.io/developer/how_does_py5_work.html

Question: Is it clear? Anything confusing? Does py5 work as you thought it did?

My goal is that folks can learn more about how py5 works and gain some insight into how to use it well. Also, interested people can use this to learn more to position themselves to maybe make bigger PRs in the future.

This kind of documentation is important for the growth of the project. There's too much info only in my head that needs to get out and be written down.

**Comment by Maltiec** [comment]
A very important info! May be some additional illustrations is important too.  I am going to make some youtube videos about basic steps in py5 (just tired of telling students the same things every 6 months), like installing, making sketches and so on. I also wanted to ask you about history of making py5 (from an idea to practical realisation) it can really make videos interesting. Btw some of my students are asking "why not to use vanilla Processing instead of py5?" - so pointing out advances of using both vanilla Processing libs and the whole bunch of CPython libs is also a part of my everyday work.)

**Reply by hx2A** [reply]
> May be some additional illustrations is important too.

Yes, some illustrations would be great. I'm not sure how to draw all of this though!

> I am going to make some youtube videos about basic steps in py5 (just tired of telling students the same things every 6 months), like installing, making sketches and so on.

I have a half baked idea that I will make a py5 youtube channel one day. It could be fun and interesting to work on, and would be a great way to promote py5.

> I also wanted to ask you about history of making py5

A big part of it is that I started working in March of 2020, during the start of covid lockdowns. Writing Python code for me is a way to alleviate anxiety. There was a lot of anxiety that year...I doubt py5 would exist had I not had that experience.

> so pointing out advances of using both vanilla Processing libs and the whole bunch of CPython libs is also a part of my everyday work.

Yes, hybrid programming is a wonderful thing! If you want to see what is coming for the next release, checkout out these two pages:

http://py5dev.ixora.io/content/hybrid_programming.html
http://py5dev.ixora.io/content/processing_mode.html

**Reply by Maltiec** [reply]
Ok, I'll try to make some experiments on illustrating how py5 works and why it is "not just another python processing mode". I'll write some questions in a week or too (I am like a squirrel in a wheel for almost a month, but I hope that on the May holidays I'll have some free time to think).

**Reply by hx2A** [reply]
> Ok, I'll try to make some experiments on illustrating how py5 works

I tried to make an illustration with all 8 items outlined in the How py5 works page but it was ridiculous. Maybe breaking it up into multiple illustrations is a better way to go.

> I am like a squirrel in a wheel for almost a month

I know the feeling!

**Reply by Maltiec** [reply]
I am thinking of a highway metaphor, with a one-way and two-way traffic. Your hybrid programming thoughts gave me an idea of using a tram or bus metaphor for buffers. I am still in a process of thinking. )
